spring-integration-test-support/src/main/java/org/springframework/integration/test/util/SocketUtils.java

```diff
@@ -28,15 +28,20 @@
 import org.springframework.util.Assert;
 
 /**
- * Contains several socket-specific utility methods. For example, you may have
- * test cases that require an open port. Rather than hard-coding the relevant port,
- * it will be better to use methods from this utility class to automatically select
- * an open port, therefore improving the portability of your test-cases across
- * systems.
+ * Contains several socket-specific utility methods. For example, you may have test cases
+ * that require an open port. Rather than hard-coding the relevant port, it will be better
+ * to use methods from this utility class to automatically select an open port, therefore
+ * improving the portability of your test-cases across systems.
+ *
+ * @deprecated - it's generally better to set the server port to 0; let the operating
+ * system choose a port; wait until the server starts (see TestingUtilities in the ip
+ * module for an example), then set the port on the client factory.
+ *
  * @author Gunnar Hillert
  * @author Gary Russell
  * @since 2.2
  */
+@Deprecated
 public final class SocketUtils {
```

Comment:
Why just don't remove it?
Otherwise it will wait until `6.0`...

Reply:
I don't think it hurts to leave it there until 6.0; we're not using it. I think it's too close to GA to remove it now (and deprecate in 4.3.x).